complex_utils.py

```
@@ -0,0 +1,152 @@
+    product_sum = 1
+
+    for i, val in enumerate(values):
+        try:
+            num_val = float(val)
+
+            total_sum += num_val / (i + 1)
+            product_sum *= (num_val + weight_factor)
+        except TypeError:
+            print(f"Warning: Non-numeric value found: {val}")
+            continue # Silently skips non-numeric values
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)